cpu/xmc1000/include/periph/usic_defs.h

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2015 Sebastian Sontberg <sebastian@sontberg.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup         cpu_xmc1000
+ * @{
+ *
+ * @file
+ * @brief           Code shared by peripheral drivers using the USIC
+ *
+ * @author          Sebastian Sontberg <sebastian@sontberg.de>
+ *
+ * @}
+ */
+
+#ifndef USIC_DEFS_H_
+#define USIC_DEFS_H_
+
+#include "periph_cpu.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup cpu_xmc1000_usic USIC configuration
+ * @ingroup cpu_xmc1000
+ * @{
+ */
+
+#define USIC_CTQIN_PDIV  (0) /**< f<sub>CTQIN</sub> is provided by f<sub>PDIV</sub> */
+#define USIC_CTQIN_PPP   (1) /**< f<sub>CTQIN</sub> is provided by f<sub>PPP</sub>  */
+#define USIC_CTQIN_SCLK  (2) /**< f<sub>CTQIN</sub> is provided by f<sub>SCLK</sub> */
+#define USIC_CTQIN_MCLK  (3) /**< f<sub>CTQIN</sub> is provided by f<sub>MCLK</sub> */
+
+#define USIC_MODE_ASC    (0) /**< select ASC/UART mode */
+#define USIC_MODE_SSC    (1) /**< select SSC/SPI mode  */
+
+/**
+ * @brief Struct to hold the configuration values necessary to
+ *        initialize a USIC channel 
+ * @{
+ */
+typedef struct {
+    uint32_t ccr;      /**< channel control register           */
+    uint32_t sctr;     /**< shift control register             */
+    uint32_t tcsr;     /**< transmit control/status register   */
+    uint32_t pcr;      /**< protocol control register          */
+    uint32_t inpr;     /**< interrupt node pointer register    */
+    uint8_t dx2_dsel;  /**< data selection for input stage DX2 */
+} usic_controls_t;
+/** @} */
+
+/**
+ * @brief   USIC BRG (baud rate generator) register configuration
+ * @{
+ */
+typedef struct {
+    union {
```

Comment:
This might be a subjective opinion, but I would prefer not to use bit-addressable register definition. Using these increases code size quite a bit compared to direct access using bitfield defines, e.g. 
```
typedef struct {
    REG32 brg;
    REG32 more_regs...;
} usic_t;

#define USIC_BRG_CLKSEL_POS     (0U)
#define USIC_BRG_CLKSEL(clk)     (clk << USIC_BRG_CLKSEL_POS)
#define USIC_BRG_TMEN      (0x00000008)
...

usic_t *usic;
usic->brg = USIC_BRG_CLKSEL(2) | USIC_BRG_TMEN;
```